core/src/test/java/ai/timefold/solver/core/testdomain/mixed/singleentity/TestdataMixedValueListener.java

```diff
@@ -0,0 +1,44 @@
+package ai.timefold.solver.core.testdomain.mixed.singleentity;
+
+import ai.timefold.solver.core.api.domain.variable.VariableListener;
+import ai.timefold.solver.core.api.score.director.ScoreDirector;
+
+import org.jspecify.annotations.NonNull;
+
+public class TestdataMixedValueListener implements VariableListener<TestdataMixedSolution, TestdataMixedValue> {
```

Comment:
Not a value listener, it is a variable listener.
Also, I'd prefer we not use any new variable listeners - we have DSVs now, let's use them.